### PR TITLE
Refine usage of daal namespaces in examples

### PR DESCRIPTION
**Pull request by PovelikinRostislav**
`daal::data_management` entities are used in the C++ samples, but they are not injected explicitly.
`service.h` header provides such injection, but its content is:

> Auxiliary functions used in C++ examples

Attempts to inject the sample code into standalone code will fail in case users won't need `service.h` header and utilities.

### CODE REVIEW

**Comment by ntfshard** [comment]
make test

**Comment by SmirnovEgorRu** [comment]
Thank you for contribution. It affects all examples not only k-means

**Comment by PovelikinRostislav** [comment]
> make test

I didn't get what do you mean. Do you mean to create a reproducer?

**Comment by averbukh** [comment]
Yes, the solution is get rid of `using namespace` in header file, while it's ok to have them in cpp files. @PovelikinRostislav, is it work for you to extend your pull request for that?

**Comment by PovelikinRostislav** [comment]
@averbukh, sounds OK for me, but I need time to test samples. I'll be quiet busy for next two weeks.

**Comment by averbukh** [comment]
Hi Rostislav,

Do you need any help to follow-up with your PR?

**Comment by PovelikinRostislav** [comment]
Hi @averbukh,

Sorry for high latency - I first started to changing the code several months ago, building the DAAL library from the scratch to test changes, but it took so long to build the library (at least whole night just to compile on my laptop), facing plenty of issues with environment so I had to postpone the activity.

What's your vision on the way to contribute into DAAL?
For this PR I used pre-built DAAL version to test, but it won't work if I had to change even a line in the library itself

I've updated a PR with corresponding changes, please review it.

**Comment by averbukh** [comment]
_Command for bot_
make test

**Comment by averbukh** [comment]
> Hi @averbukh,
> 
> Sorry for high latency - I first started to changing the code several months ago, building the DAAL library from the scratch to test changes, but it took so long to build the library (at least whole night just to compile once on my laptop), facing plenty of issues with environment so I had to postpone it.
> 
> What's your vision on the way to start contributing into DAAL? Is there any easier way to jump into?
> How much time does it take on average to compile the project on laptop?
> 
> For this PR I used pre-built DAAL version to test, but it won't work if I had to change even a line in the library itself.
> 
> I've updated a PR with corresponding changes, please review it.

We are working on build time improvement, but it isn't something you can use just now. I recommend to use build for only one CPU (see detail [here](https://software.intel.com/en-us/articles/intel-daal-building-the-customized-set-of-selected-algorithms-and-cpu-optimizations)). The time depends on your OS. For example, compilation for MacOS X took ~30 minutes using following make parameters: `make daal -j4 PLAT=mac32e COMPILER=clang REQCPU="avx2"`

File: examples/cpp/source/utils/service.h [hunk 1]
```diff
@@ -148,7 +146,7 @@ CSRNumericTable * createSparseTable(const std::string & datasetFileName)
     size_t * resultRowOffsets      = NULL;
     size_t * resultColIndices      = NULL;
     item_type * resultData         = NULL;
-    CSRNumericTable * numericTable = new CSRNumericTable(resultData, resultColIndices, resultRowOffsets, nFeatures, nVectors);
+    auto numericTable = new daal::data_management::CSRNumericTable(resultData, resultColIndices, resultRowOffsets, nFeatures, nVectors);
```
[the remark this answers is not in the document]
Comment (PovelikinRostislav) [reply examples/cpp/source/utils/service.h]: Done

**Comment by PovelikinRostislav** [comment]
@averbukh, do I need to put 

> make test

every time I upload meaningful changes?

**Comment by napetrov** [comment]
@PovelikinRostislav   It's required prior to merging changes in repo

**Comment by averbukh** [comment]
> @averbukh, do I need to put
> 
> > make test
> 
> every time I upload meaningful changes?

No, the bot accepts this command from limited number of people. Sorry. It's can be done after some short specific review only. See [CONTRIBUTING](https://github.com/intel/daal/blob/master/CONTRIBUTING.md) for details about public and private CI.

**Comment by PovelikinRostislav** [comment]
@averbukh, what should I do to fix private CI?

**Comment by napetrov** [comment]
restart

**Comment by napetrov** [comment]
Previous failures have been fixed. ci will fail due to new added tests but i see no problems merging it now

**Comment by averbukh** [comment]
_Command for bot_
make test

**Pull request merged**